### PR TITLE
Kernel directory file cleanup

### PR DESCRIPTION
**Pull request by jakewins**
Removed some unused files in the kernel module directory, and updated the readme.

### CODE REVIEW

File: community/server/pom.xml [hunk 1]
```diff
@@ -62,7 +62,7 @@
     <webdriver.version>2.42.2</webdriver.version>
     <webdriver.implementation>Firefox</webdriver.implementation>
 
-    <!-- Default location for chrome controller binary, used by webdriver for running tests in Chrome. Available from (http://code.google.com/p/chromium/downloads/list) see README.txt for details on this. -->
+    <!-- Default location for chrome controller binary, used by webdriver for running tests in Chrome. Available from (http://code.google.com/p/chromium/downloads/list) see README.md for details on this. -->
```
Comment (chrisvest) [inline community/server/pom.xml]: What readme is this referring to?

Comment (jakewins) [inline community/server/pom.xml]: Ugh, stupid intellij refactoring

**Comment by chrisvest** [comment]
@jakewins The build failure is something unrelated. Did you meant to squash the two commits?

**Comment by tinwelint** [comment]
The test failure is a known flaky test... I'm rerunning.

File: community/kernel/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ - The embedded Java API
+    - org.neo4j.graphdb
+ - The embedded Java API implementation
+    - org.neo4j.kernel.coreapi
+    - org.neo4j.kernel.core
+ - The embedded Traversal Java API
+   - org.neo4j.graphdb.traversal
+ - The embedded Traversal API implementation
+    - org.neo4j.kernel.traversal
+ - Batch Import
```
Comment (tinwelint) [inline community/kernel/README.md:14]: let's not confuse the term `Batch Import` with the old `BatchInserter`, also both could be mentioned:

```
- Batch insertion
  - org.neo4j.unsafe.batchinsert
- Batch import
  - org.neo4j.unsafe.impl.batchimport
```

the manual mentions how they relate to and differentiate from one another.

**Comment by jakewins** [comment]
@tinwelint updated to separate legacy batch importer and new batch insert
@chrisvest +1, squashed to one commit

If either of you wouldn't mind merging if you're happy, that'd be awesome!

**Pull request merged**